tools/git-change-exec/actions-lint.go

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type lintSpdx struct {
+	extsMap    map[string]func(path string)
+	pathsToFix []string
+}
+
+func (s *lintSpdx) dontfix(path string) {
+	log.Printf("Cannot fix %s ...\n", path)
+}
+
+func (s *lintSpdx) yamlfix(path string) {
+	license := []string{
+		"# Copyright (c) 2024 Zededa, Inc.\n",
+		"# SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func (s *lintSpdx) gofix(path string) {
+	license := []string{
+		"// Copyright (c) 2024 Zededa, Inc.\n",
+		"// SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func (s *lintSpdx) dockerfilefix(path string) {
+	license := []string{
+		"# Copyright (c) 2024 Zededa, Inc.\n",
+		"# SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func prepend(path string, license []string) {
+	backupFh, err := os.CreateTemp("/var/tmp", "git-change-exec-spdx-fix")
+	if err != nil {
+		log.Fatalf("could not create temp file: %v", err)
+	}
+	backupPath := backupFh.Name()
+	defer os.Remove(backupPath)
+
+	for _, line := range license {
+		fmt.Fprint(backupFh, line)
+	}
+
+	origFh, err := os.Open(path)
+	if err != nil {
+		log.Fatalf("could not open original file %s: %v", path, err)
+	}
+
+	_, err = io.Copy(backupFh, origFh)
+	if err != nil {
+		log.Fatalf("could not copy: %v", err)
+	}
+
+	backupFh.Close()
+	origFh.Close()
+
+	err = copyFile(backupPath, path)
+	if err != nil {
+		log.Fatalf("could not rename %s -> %s: %v", backupPath, path, err)
+	}
+
+}
+
+func (s *lintSpdx) init() {
+	s.extsMap = map[string]func(path string){
+		".sh":        s.dontfix,
+		".go":        s.gofix,
+		".c":         s.dontfix,
+		".h":         s.dontfix,
+		".py":        s.dontfix,
+		".rs":        s.dontfix,
+		".yaml":      s.yamlfix,
+		".yml":       s.yamlfix,
+		"Dockerfile": s.dockerfilefix,
+	}
+
+	s.pathsToFix = make([]string, 0)
+}
+
+func (s *lintSpdx) hasSpdx(path string) bool {
+	fh, err := os.Open(path)
+	if err != nil {
+		log.Fatalf("could not open %s: %v", path, err)
+	}
+
+	defer fh.Close()
+
+	licenseNames := []string{
+		"Apache-2.0",
+		"MIT",
+		"BSD-2-Clause",
+		"BSD-3-Clause",
+		"ISC", // Internet Systems Consortium license
+		"Python-2.0",
+		"PostgreSQL",
+		"Zlib",
+		"BSL-1.0", // Boost Software License 1.0
+		"CC0-1.0", // Creative Commons Zero v1.0 Universal
+	}
+	identifierFound := false
+	licenseFound := false
+	copyRightFound := false
+	scanner := bufio.NewScanner(fh)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "SPDX-License-Identifier") {
+			identifierFound = true
+		}
+
+		for _, ln := range licenseNames {
+			if strings.Contains(line, ln) {
+				licenseFound = true
+				break
+			}
+		}
+
+		if strings.Contains(line, "Copyright (c) ") {
```

Comment:
It should be less strict, as we have already figured out on one of your PRs:
"Copyright *(c) "